atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/futures/FuturesCombiner.java

```diff
@@ -14,26 +14,20 @@
  * limitations under the License.
  */
 
-package com.palantir.atlasdb.keyvalue.cassandra.async.futures;
+package com.palantir.atlasdb.futures;
 
 import java.util.Map;
 import java.util.Optional;
+import java.util.concurrent.ExecutorService;
 
 import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.processors.AutoDelegate;
+
+@AutoDelegate
+public interface FuturesCombiner extends AutoCloseable {
```

Comment:
I still think it's helpful to briefly outline what the role of this class is in terms of docs.

Reply:
Added an explanation would appreciate another look.